crates/anstyle-roff/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+        roff_docs.push(as_roff(&styled))
+    }
+
+    let mut doc = Roff::new();
+    doc.extend(roff_docs);
+    doc
+}
+
+fn as_roff(styled: &StyledStr) -> Roff {
+    let style = styled.style;
+    let mut doc = Roff::new();
+    doc.extend([
+        set_color((&style.get_fg_color(), &style.get_bg_color())),
+        set_effects(styled),
+    ]);
+    doc
+}
+
+fn set_effects(styled: &StyledStr) -> Roff {
+    // Roff (the crate) only supports these inline commands
+    //  - Bold
+    //  - Italic
+    //  - Roman (plain text)
+    // If we want more support, or even support combined formats, we will need
+    // to push improvements to roff upstream or implement a more thorough roff crate
+    // perhaps by spinning off some of this code
+    let effects = styled.style.get_effects();
+    let mut doc = Roff::new();
+
+    if effects.contains(anstyle::Effects::BOLD) {
+        doc.text(vec![bold(styled.text)]);
+        return doc;
+    }
+
+    if effects.contains(anstyle::Effects::ITALIC) {
+        doc.text(vec![italic(styled.text)]);
+        return doc;
+    }
+
+    if effects.is_plain() {
+        doc.text(vec![roff::roman(styled.text)]);
+        return doc;
+    }
+
+    doc
+}
+
+/// Set the foreground, background color
+fn set_color(colors: (&Option<Color>, &Option<Color>)) -> Roff {
+    let mut doc = Roff::new();
+    // Set foreground
+    add_color_to_roff(&mut doc, control_requests::FOREGROUND, colors.0);
+    // Set background
+    add_color_to_roff(&mut doc, control_requests::BACKGROUND, colors.1);
+    doc
+}
+
+fn add_color_to_roff(doc: &mut Roff, control_request: &str, color: &Option<Color>) {
+    match color {
+        Some(Color::Rgb(c)) => {
+            let name = rgb_name(c);
+            doc.control(
+                control_requests::CREATE_COLOR,
+                vec![name.as_str(), "rgb", as_hex(c).as_str()],
+            )
+            .control(control_request, vec![name.as_str()]);
+        }
+
+        Some(Color::Ansi(c)) => {
+            doc.control(control_request, vec![ansi_color_to_roff(c)]);
+        }
+        _ => {
+            // TODO: get rid of "default" hardcoded str?
+            doc.control(control_request, vec!["default"]);
+        }
+    }
+}
+
+fn rgb_name(c: &RgbColor) -> String {
+    format!("hex_{}", as_hex(c).as_str())
+}
+
+fn as_hex(rgb: &RgbColor) -> String {
+    let val: usize = ((rgb.0 as usize) << 16) + ((rgb.1 as usize) << 8) + (rgb.2 as usize);
+    format!("#{:06x}", val)
+}
+
+fn ansi_color_to_roff(color: &anstyle::AnsiColor) -> &'static str {
+    match color {
+        anstyle::AnsiColor::Black => "black",
+        anstyle::AnsiColor::Red => "red",
+        anstyle::AnsiColor::Green => "green",
+        anstyle::AnsiColor::Yellow => "yellow",
+        anstyle::AnsiColor::Blue => "blue",
+        anstyle::AnsiColor::Magenta => "magenta",
+        anstyle::AnsiColor::Cyan => "cyan",
+        anstyle::AnsiColor::White => "white",
+        _ => "default",
```

Comment:
I feel like `Bright` should either be converted into the color or converted into bold+color, see #12

Reply:
Done. Bright now maps to the colors, and there is extra logic when applying the Bold Effect to check if the Color is a Bright variant.